src/daft-micropartition/src/partitioning.rs

```diff
@@ -0,0 +1,208 @@
+use std::{collections::HashMap, sync::Arc};
+
+use common_error::{DaftError, DaftResult};
+use daft_dsl::Expr;
+use futures::stream::BoxStream;
+
+use crate::MicroPartition;
+type PartId = String;
+
+#[derive(Debug, Clone)]
+pub struct Boundaries {
```

Comment:
Is boundaries the best name here? kinda confused why it is called boundaries

Reply:
The naming here was copied over from the python implementation. I wanted to make as much a 1-1 as possible.

Reply:
ok maybe link to it if possible in doc comment or mention it?